internal/controllers/v1beta1/gateway/gateway.go

```diff
@@ -140,11 +140,18 @@ func (c *GatewayController) Reconcile(ctx context.Context, request reconcile.Req
 func (c *GatewayController) CreateCertificate(ctx context.Context, gateway *networkingv1beta1.Gateway, server *v1beta1.Server) error {
 	log := log.FromContext(ctx)
 	issuer := c.clusterIssuer
+	var tcb bool
```

Comment:
nit: this variable is used throughout the function, its name should be more descriptive.

```golang
	var issueTempCertificate bool
```